### PR TITLE
Init

### PR DESCRIPTION
**Pull request by dylanaraps**
The distribution won't package anything beyond s6 and sinit (what's in this PR) but baseinit will support many more 
(as well as device managers and service managers, etc) (send suggestions). Still a lot of work to do but this PR should give 
those wanting to try a functional system. 

Look at the new README files and then look at https://github.com/kisslinux/init/blob/master/etc/rc.conf for more
info. Basically, you use the alternatives system to swap things around and then set the corresponding values in
/etc/rc.conf. 

Edit: Documentation can be found here: https://github.com/kisslinux/init

### CODE REVIEW

File: extra/dhcpcd/build [hunk 1]
```diff
@@ -13,4 +13,3 @@ make install
 # Install runit service.
 mkdir -p "$1/etc/sv/dhcpcd"
 cp -f dhcpcd.run "$1/etc/sv/dhcpcd/run"
-ln -sf /run/runit/supervise.dhcpcd "$1/etc/sv/dhcpcd/supervise"
```
Comment (jedahan) [inline extra/dhcpcd/build:16]: Does this mean users need to manually symlink both `/run/runit...` and `/var/service...` to enable services using the default (busybox) init?

Comment (dylanaraps) [reply extra/dhcpcd/build:16]: No. The supervise directory is automatically created if it does not exist. This change merely stops supervise from
being a symbolic link to `/run` (this is an optional configuration). So, either supervise will be created by the service manager or the user can setup the symbolic links manually (if they want to do this).

Comment (jedahan) [reply extra/dhcpcd/build:16]: excellent

File: extra/s6/version [hunk 2]
```diff
@@ -0,0 +1 @@
+2.10.0.3 1
```
Comment (Rio6) [inline extra/s6/version:1]: We might want to bump it to 2 so people using s6 from the community repo gets updated.

Comment (dylanaraps) [reply extra/s6/version:1]: Sure thing.

**Comment by dylanaraps** [comment]
Rethinking things some. This model of juggling behaviors between components is bound to get complex/unmaintainable as support is expanded.  It also creates a rigid structure around observed behavior and ends up *limiting* users. Instead, I want 
to support/provide one default configuration leaving the possibility open for users to swap to something else. This means 
going back to the drawing board (doing more work on this PR + baseinit) and making s6 the default in the future.

**Comment by Rio6** [comment]
I agree that having one framework would make everything complex in the future, especially that different inits/service managers have different way to run all the mounting, mdev, etc, commands. Like s6-linut-init has /etc/s6-linux-init/current/scripts/*, runit has /etc/runit/{1,2,3}, s6-rc uses oneshot services. I think what KISS has right now works great for different inits and service managers already. 

One idea I have is to have a package that provides all the common functionalities a init might need - like mounting special dirs, fstab, remounting root, setting up hostname, setting sysctl, loading modules, loading console keymaps... And for each init they just need to call those scripts from their preferred locations (set up by the maintainers or the users). This way we can reduce the amount of scripts needed for each unit while having more flexibility.

**Comment by dylanaraps** [comment]
Will open another PR shortly.

**Pull request closed without merging**